gfx/gfx_thumbnail_path.c

```diff
@@ -555,6 +566,174 @@ bool gfx_thumbnail_set_content_playlist(
+      content_name_no_ext = path_remove_extension(tmp_buf);
+      if (!content_name_no_ext)
+         content_name_no_ext = tmp_buf;
+
+      gfx_thumbnail_fill_content_img(path_data->content_img_full,
+         sizeof(path_data->content_img_full), content_name_no_ext,false);
+      gfx_thumbnail_fill_content_img(path_data->content_img,
+         sizeof(path_data->content_img), path_data->content_label,false);
+      /* Explicit zero if full name is same as standard name - saves some queries later. */
+      if(strcmp(path_data->content_img, path_data->content_img_full) == 0) 
+      {
+         path_data->content_img_full[0] = '\0';
+      }
+      gfx_thumbnail_fill_content_img(path_data->content_img_short,
+         sizeof(path_data->content_img_short), path_data->content_label,true);
+   }
+
+   /* Store playlist index */
+   path_data->playlist_index = idx;
+
+   /* Redundant error check... */
+   if (string_is_empty(path_data->content_img))
+      return false;
+
+   /* Thumbnail image name is done -> now check if
+    * per-content database name is defined */
+   if (string_is_empty(db_name))
+      playlist_get_db_name(playlist, idx, &db_name);
+   if (!string_is_empty(db_name))
+   {
+      /* Hack: There is only one MAME thumbnail repo,
+       * so filter any input starting with 'MAME...' */
+      if (strncmp(db_name, "MAME", 4) == 0)
+      {
+         path_data->content_db_name[0] = path_data->content_db_name[2] = 'M';
+         path_data->content_db_name[1] = 'A';
+         path_data->content_db_name[3] = 'E';
+         path_data->content_db_name[4] = '\0';
+      }
+      else
+      {
+         char *db_name_no_ext = NULL;
+         char tmp_buf[PATH_MAX_LENGTH];
+         const char* pos = strchr(db_name, '|');
+
+         if (pos && (size_t) (pos - db_name)+1 < sizeof(tmp_buf)) {
+            /* If db_name comes from core info, and there are multiple 
+             * databases mentioned separated by |, use only first one */
+            strlcpy(tmp_buf, db_name, (size_t) (pos - db_name)+1);
+         }
+         else {
+            /* Remove .lpl extension
+             * > path_remove_extension() requires a char * (not const)
+             *   so have to use a temporary buffer... */
+            strlcpy(tmp_buf, db_name, sizeof(tmp_buf));
+         }
+         db_name_no_ext = path_remove_extension(tmp_buf);
+
+         if (!string_is_empty(db_name_no_ext))
+            strlcpy(path_data->content_db_name,
+                  db_name_no_ext, sizeof(path_data->content_db_name));
+         else
+            strlcpy(path_data->content_db_name,
+                  tmp_buf, sizeof(path_data->content_db_name));
+      }
+   }
+
+   gfx_thumbnail_update_path(path_data, GFX_THUMBNAIL_ICON);
+
+   /* Playlist entry is valid -> it is now 'safe' to
+    * extract any remaining playlist metadata
+    * (i.e. thumbnail display modes) */
+   path_data->playlist_icon_mode = PLAYLIST_THUMBNAIL_MODE_DEFAULT;  //PLAYLIST_THUMBNAIL_MODE_LOGO;
```

Comment:
Do not use C++ comments in C source files please. This breaks C89 builds.

Reply:
Comments removed.